### PR TITLE
Remove override for eye and add override for eye_out.

### PR DESCRIPTION
**Pull request by ailzhang**

### CODE REVIEW

File: scripts/gen.py [hunk 1]
```diff
@@ -99,6 +99,7 @@ class ArgTemplate(string.Template):
     'hamming_window',
     'hann_window',
     'empty_like',
+    'eye',
```
Comment (dlibenzi) [inline scripts/gen.py]: Maybe we can alphabetically order the ones below the FIXME comment?

File: torch_xla/csrc/aten_xla_type.cpp [hunk 2]
```diff
@@ -1145,17 +1145,18 @@ at::Tensor& AtenXlaType::expm1_(at::Tensor& self) {
...
   XLA_FN_COUNTER("xla::");
-  return eye(n, n, options);
+  XLATensor out_tensor = bridge::GetXlaTensor(out);
+  XLATensor::eye_out(out_tensor, n, n);
+  return out;
 }
 
-at::Tensor AtenXlaType::eye(int64_t n, int64_t m,
-                            const at::TensorOptions& options) {
+at::Tensor& AtenXlaType::eye_out(at::Tensor& out, int64_t n, int64_t m) {
```
Comment (dlibenzi) [inline torch_xla/csrc/aten_xla_type.cpp]: Do we need to do this?
It seems pretty confusing.
Sometimes overriding `foo()` is enough. Sometimes you need `foo()` and `foo_out()`. Sometimes only `foo_out()`.
Is there a random number generator to drive these decisions on the PT side? 😁

File: torch_xla/csrc/tensor_methods.cpp [hunk 3]
```diff
@@ -953,6 +953,16 @@ XLATensor XLATensor::eye(xla::int64 lines, xla::int64 cols,
       device, element_type);
 }
 
+void XLATensor::eye_out(XLATensor& out, xla::int64 lines, xla::int64 cols) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: Is the `eye()` above still used?

File: torch_xla/csrc/tensor_methods.cpp [hunk 4]
```diff
@@ -953,6 +953,16 @@ XLATensor XLATensor::eye(xla::int64 lines, xla::int64 cols,
...
+  if (cols < 0) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_methods.cpp]: In these cases, I tend to like more: `cols >= 0 ? cols : lines` directly as function argument below.

File: scripts/gen.py (hunk 1 above)
Comment (ailzhang) [reply scripts/gen.py]: fixed! ;)

File: torch_xla/csrc/tensor_methods.cpp (hunk 4 above)
Comment (ailzhang) [reply torch_xla/csrc/tensor_methods.cpp]: Haha in `eye`'s case, `eye_out` is the leaf node (where we dispatch to different devices). 
Yea PT is not consistent in this case, our integration will help clean up some in the future ;)

Comment (ailzhang) [reply torch_xla/csrc/tensor_methods.cpp]: oops wrong place for the reply :P 
Yea I'll fix the new comments :D

File: torch_xla/csrc/tensor_methods.cpp (hunk 3 above)
Comment (ailzhang) [reply torch_xla/csrc/tensor_methods.cpp]: Yep it's still used here. https://github.com/pytorch/xla/blob/master/torch_xla/csrc/tensor_ops.cpp#L80

**Pull request merged**